src/components/AgeTabPicker.tsx

```diff
@@ -0,0 +1,81 @@
+import { useRef, useLayoutEffect } from 'react';
+import { TabList, Tab, TabStateReturn } from 'reakit';
+
+type TrainAge =
+    | { key: 'old_vehicles'; label: 'Old' }
+    | { key: 'new_vehicles'; label: 'New' }
+    | { key: 'vehicles'; label: 'All' };
+
+const trainTypes: TrainAge[] = [
+    { key: 'old_vehicles', label: 'Old' },
+    { key: 'new_vehicles', label: 'New' },
+    { key: 'vehicles', label: 'All' },
+];
+
+interface AgeTabPickerProps {
+    tabState: TabStateReturn;
+    tabColor: string;
+}
+
+export const AgeTabPicker: React.FC<AgeTabPickerProps> = ({ tabState, tabColor }) => {
```

Comment:
This component duplicates a fair amount of the logic in the existing `TabPicker`. Could we make that component a bit more generic and then reuse it for a `LineTabPicker` and an `AgeTabPicker`?

Reply:
They share some logic, but now the `useLayoutEffect`s are different, as well as how it generates the tablist. We could extract out some logic, but I think we'd end up with two files with similar lengths and complexities. I might say put a pin in it (make an issue) if we care about making it more generic.